### ISSUE TITLE
Some question about the benchmark

### ISSUE THREAD

**Issue by sunset1995** [kwea123/ngp_pl#3]
Thanks for your great work! :)
The code is really easy to follow and understand. I love it!

Just some questions.

1. What does the FPS means in the table? Is it the inference time of a `800x800` images?
2. Is the reported 60 FPS of the official instant-ngp is c++/cuda run on RTX 2080Ti as well?
3. All the reported 3 implementations are mentioned to be trained for 5 minutes. Does it means that the faster one is actually trained for more iterations?

**Comment by kwea123** [comment]
Hi, I also learnt a lot from your dvgo!

1. It's the inference speed (frames per second) of an image in its original resolution (i.e. 800x800 for Synthetic-NeRF, 1920x1080 for TanksAndTemple, etc), averaged for all testing images.
2. It's the number reported in the paper, I haven't tried benchmarking on my machine. If you have the number, you are welcomed to share!
3. Theoretically yes, I think instant-ngp paper trains more iterations than the pytorch versions. Mine and torch-ngp actually has the same number of iterations by default (mine is 20k steps * 8192 rays/step and torch-ngp is 40k steps * 4096 rays/step). Also, the "5 minute" is not exact, some is trained for 5m53s (e.g. ship) and some is trained for 4m46s (e.g. ficus); in average each scene finishes around 5 minutes for this default number of iterations.

**Comment by kwea123** [comment]
I will upload a jupyter notebook that allows for measuring the time, the benchmark scripts only train all scenes, the timing is not exact because it includes time to save images to the disk..

**Comment by kwea123** [comment]
The notebook `test.ipynb` and a pretrained model is uploaded, you can use them to evaluate and measure the fps.

**Comment by kwea123** [comment]
I just noticed a big bottleneck in data loading, now I fixed it, and the default config finishes around only 3 minutes now... I will re-benchmark for 5 minutes training... Exhausting but very happy! :smile:

**Comment by sunset1995** [comment]
That's really great~!

One more question. Do you have any idea about the PSNRs difference comparing to the official instant-ngp?
It seems that the reproduced PSNRs do not match the official reported PSNRs.

**Comment by sunset1995** [comment]
Not sure whether the gradient scaling ([intent of LOSS_SCALE](https://github.com/NVlabs/instant-ngp/discussions/453)) is necessary or is already handled by torch lightning.

**Comment by sunset1995** [comment]
Some other differences I found:
- They apply L2 regularization to the network weight.
- They use `betas=(0.9, 0.99)`, `eps=1e-15` for Adam. Quote: "small value of 𝜖 = 1e−15 can significantly accelerate the convergence of the hash table entries when their gradients are sparse and weak."
- Their learning annealing is similar to `MultiStepLR`.

**Comment by kwea123** [comment]
Thank you, let me try these tricks. I was wondering whether training longer makes the result better, but it didn't. On lego scene training longer can only achieve around 35.60 PSNR, still far from their 36.39

**Comment by kwea123** [comment]
@sunset1995 I tried the tricks but they don't change anything...
Btw, I tried training with official code with the following steps, it seems on my 2080Ti, PSNR 35.72 is what I can get within 5 minutes.

I tried running `srcipts/run.py`, but by default it loads all data for training:
![Screenshot from 2022-07-06 13-34-21](https://user-images.githubusercontent.com/11364490/177469030-ad630104-087d-4e26-9604-c791816605f3.png)
and the testing PSNR is of course high. I had to **manually** remove the `val.json` and `test.json` in order to train on the `train` split only. After doing that and training for 5 minutes, however, I don't get as good results as the paper. The PSNR is only 35.72
![Screenshot from 2022-07-06 13-36-58](https://user-images.githubusercontent.com/11364490/177469314-6ae68d91-0cae-41a0-9f39-5727a91b0fde.png)

**Comment by kwea123** [comment]
Just noticed another significant difference, in the original code the scene is compressed to [0, 1], where here (and torch-ngp) compress to **[-1, 1]**. We all adapted the step size, etc according to scale, so it's OK, but we both forgot to change the `grid_size` of `density_grid`... it is still 128 instead of 256. I just did a quick experiment that compresses the scene to [-0.5, 0.5] instead of [-1, 1], and the quality increases a lot now. Maybe I will make this the default.

**Comment by sunset1995** [comment]
That awesome!

**Comment by cubantonystark** [comment]
Hello and thanks for your contribution. This is an amazing leap forward in NeRF technology even though you set it as an experiment. I have been following your implementations for a while now and you don't cease to amaze. I installed ngp_pl but I'm running into this issue.

My run is killed at 98%. Is this related to OOM?

Epoch 0:  72%|██████████████████████████████████████████████████████████████████████████████████████▉                                  | 718/1000 [00:13<00:05, 52.67it/s, loss=0.00166, v_num=0, train/psnr=27.50]
Epoch 29:  98%|███████████████████████████████████████████████████████████████████████████████████████████████████████████████████   | 1170/1200 [04:17<00:06,  4.54it/s, loss=0.000224, v_num=0, train/psnr=36.20]
Killedtion DataLoader 0:  85%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████▉                      | 170/200 [03:53<22:22, 44.76s/it]

**Comment by kwea123** [comment]
@cubantonystark This is an unrelated problem. Can you open in a new issue?
The `Killed` message generally comes from OOM (RAM), can you add `--no_save_test` argument and train again? Then you can use the `test.ipynb` to generate test images later.

**Comment by kwea123** [comment]
I think I roughly attained instant-ngp quality, close the issue for now.
